gh-69753: Add Bytes Object Support to Shlex (#22657)
python/cpython: Misc/NEWS.d/next/Library/2020-10-12-03-46-25.bpo-25567.xgfgij.rst

> Add support for bytes objects in the shlex module.

Reviewer: ```suggestion
Add support for :class:`bytes` objects to the :mod:`shlex` module.
```

You should probably explain that `join()` returns an empty string as well in the docs. This is important. Or introduce a separate function for bytes objects (which I think would be preferrable because you wouldn't have all those warnings to handle).
